File: rb/BeginnersSelection/07_KagamiMochi.rb
# https://atcoder.jp/contests/abs/tasks/abc085_b

class Object
  def let(f)
    f.call self
  end
end

gets
  .to_i
  .let(->(n) { (1..n) })
  .map { |_| gets }
  .uniq
  .size
  .let(method(:puts))

# https://atcoder.jp/contests/abs/submissions/21520385

File: py/BeginnersSelection/07_KagamiMochi.py
# https://atcoder.jp/contests/abs/tasks/abc085_b

from functools import reduce

class W:
  def __init__(self, v): self.v = v
  let = lambda self, f: W(f(self.v))
  map = lambda self, f: W(map(f, self.v))

W(input()) \
  .let(int) \
  .let(lambda n: range(0, n)) \
  .map(lambda _: input()) \
  .let(set) \
  .let(len) \
  .let(print)

# https://atcoder.jp/contests/abs/submissions/21717507
